test/gtest/unit_conv_solver.cpp

```diff
@@ -23,34 +23,15 @@
  * SOFTWARE.
  *
  *******************************************************************************/
-#pragma once
 
-#include <gtest/gtest.h>
-#include "conv_common.hpp"
-#include "get_handle.hpp"
-#include "tensor_util.hpp"
-#include <miopen/conv/data_invoke_params.hpp>
+#include "unit_conv_solver.hpp"
 
-#include <miopen/type_name.hpp>
-#include <miopen/rank.hpp>
-
-template <typename Solver, typename Context, typename Problem>
-auto GetSolutionImpl(miopen::rank<1>, Solver s, const Context& ctx, const Problem& problem)
-    -> decltype(s.GetSolution(ctx, problem, s.GetDefaultPerformanceConfig(ctx, problem)))
-{
-    return s.GetSolution(ctx, problem, s.GetDefaultPerformanceConfig(ctx, problem));
-}
-
-template <typename Solver, typename Context, typename Problem>
-auto GetSolutionImpl(miopen::rank<0>, Solver s, const Context& ctx, const Problem& problem)
-    -> decltype(s.GetSolution(ctx, problem))
-{
-    return s.GetSolution(ctx, problem);
-}
-
-template <typename Solver, typename Context, typename Problem>
-miopen::solver::ConvSolution GetSolution(Solver s, const Context& ctx, const Problem& problem)
+bool UnitTestConvSolver::CheckTestSupportedForDevice(Gpu supported_gpus)
 {
-    auto solution = GetSolutionImpl(miopen::rank<1>{}, s, ctx, problem);
-    return solution;
+    auto gpu = GetDevGpuType();
+    if(static_cast<int>(gpu) & static_cast<int>(supported_gpus))
+    {
+        return true;
+    }
+    return false;
```

Comment:
I'd suggest to pass `supported_gpu` into helper and reduce number of static casts in the tests.
```suggestion
    return IsTestSupportedByDevice(supported_gpus);
```